vendor/knative.dev/eventing/cmd/webhook/main.go

```diff
@@ -0,0 +1,214 @@
+	ctxFunc := func(ctx context.Context) context.Context {
+		return ctx
+	}
+
+	// Watch the default-ch-webhook ConfigMap and dynamically update the default
+	// Channel CRD.
+	// TODO(#2128): This should be persisted to context in the context function
+	// above and fetched off of context by the api code.  See knative/serving's logic
+	// around config-defaults for an example of this.
+	chDefaulter := defaultchannel.New(logger.Desugar())
+	eventingduckv1alpha1.ChannelDefaulterSingleton = chDefaulter
+	cmw.Watch(defaultchannel.ConfigMapName, chDefaulter.UpdateConfigMap)
+
+	return defaulting.NewAdmissionController(ctx,
+
+		// Name of the resource webhook.
+		"webhook.eventing.knative.dev",
+
+		// The path on which to serve the webhook.
+		"/defaulting",
+
+		// The resources to validate and default.
+		ourTypes,
+
+		// A function that infuses the context passed to Validate/SetDefaults with custom metadata.
+		ctxFunc,
+
+		// Whether to disallow unknown fields.
+		true,
+	)
+}
+
+func NewValidationAdmissionController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	return validation.NewAdmissionController(ctx,
+
+		// Name of the resource webhook.
+		"validation.webhook.eventing.knative.dev",
+
+		// The path on which to serve the webhook.
+		"/resource-validation",
+
+		// The resources to validate and default.
+		ourTypes,
+
+		// A function that infuses the context passed to Validate/SetDefaults with custom metadata.
+		func(ctx context.Context) context.Context {
+			// return v1.WithUpgradeViaDefaulting(store.ToContext(ctx))
+			return ctx
+		},
+
+		// Whether to disallow unknown fields.
+		true,
+	)
+}
+
+func NewConfigValidationController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	return configmaps.NewAdmissionController(ctx,
+
+		// Name of the configmap webhook.
+		"config.webhook.eventing.knative.dev",
+
+		// The path on which to serve the webhook.
+		"/config-validation",
+
+		// The configmaps to validate.
+		configmap.Constructors{
+			tracingconfig.ConfigName: tracingconfig.NewTracingConfigFromConfigMap,
+			// metrics.ConfigMapName():   metricsconfig.NewObservabilityConfigFromConfigMap,
+			logging.ConfigMapName(): logging.NewConfigFromConfigMap,
+		},
+	)
+}
+
+func NewSinkBindingWebhook(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	sbresolver := sinkbinding.WithContextFactory(ctx, func(types.NamespacedName) {})
+
+	return psbinding.NewAdmissionController(ctx,
+
+		// Name of the resource webhook.
+		"sinkbindings.webhook.sources.knative.dev",
+
+		// The path on which to serve the webhook.
+		"/sinkbindings",
+
+		// How to get all the Bindables for configuring the mutating webhook.
+		sinkbinding.ListAll,
+
+		// How to setup the context prior to invoking Do/Undo.
+		sbresolver,
+	)
+}
+
+// TODO(#2312): Remove this after v0.13.
+func NewLegacySinkBindingWebhook(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
```

Comment:
I'd question bringing this legacy just to remove it, but up to you :)

Reply:
Nack, vendor file.